modules/custom-status/custom-status.php

```diff
@@ -678,23 +705,26 @@ public function delete_custom_status( int $status_id ): bool|WP_Error {
 
 		$old_status_slug = $old_status->slug;
 
-		if ( $this->is_restricted_status( $old_status_slug ) || 'draft' === $old_status_slug ) {
-			return new WP_Error( 'restricted', __( 'Restricted status ', 'vip-workflow' ) . '(' . $old_status->name . ')' );
+		// Don't allow changing the name or slug of the Draft or pending status as they are core statuses
+		$banned_statuses = [ 'draft', 'pending' ];
+
+		if ( self::is_restricted_status( $old_status_slug ) || in_array( $old_status->slug, $banned_statuses ) ) {
```

Comment:
I wonder if we can just move `draft` (and maybe `pending`) to `is_restricted_status()`? If we always ban it, it seems clearer to do in that function.

Reply:
Given what I said [here](https://github.com/Automattic/vip-workflow-plugin/pull/59#discussion_r1792705172), I'll do that